sentry/src/main/java/io/sentry/Scopes.java

```diff
@@ -646,9 +646,7 @@ public void withScope(final @NotNull ScopeCallback callback) {
 
     } else {
       final @NotNull IScopes forkedScopes = forkedCurrentScope("withScope");
-      // TODO [HSM] should forkedScopes be made current inside callback?
```

Comment:
Yes, I'd expect Sentry.setTag to operate on the forked scopes inside `withScope`